### PR TITLE
Update to latest SDK version

### PR DESCRIPTION
**Pull request by asaj**
Updates to the latest SDK and agent versions, minor cleanup of config code.

### CODE REVIEW

**Comment by nambrot** [comment]
@yorhodes can you take a look

File: src/config.ts [hunk 1]
```diff
@@ -100,30 +106,93 @@ export function coerceAddressToBytes32(value: string): string {
...
+    mailbox: new Mailbox__factory(),
+    defaultIsmInterchainGasPaymaster: new OverheadIgp__factory(),
+  };
+  const filteredAddressesMap = filterAddressesMap(
+    addressesMap,
+    routerConfigFactories,
+  );
+  return Object.fromEntries(
+    chains.map((chain) => {
+      const routerConfig: RouterConfig = {
```
Comment (jmrossy) [inline src/config.ts:170]: Def not a blocking issue but wondering if it's worth making a util function for this `Object.fromEntries`...`.map` pattern you're using in a few places.

File: src/core/HyperlanePermissionlessDeployer.ts [hunk 2]
```diff
@@ -105,88 +104,75 @@ export class HyperlanePermissionlessDeployer {
...
-        return { ism: ism.multisigIsm.address };
-      },
+
+    // 4. Deploy ISM contracts to remote chains
+    this.logger(`Deploying ISMs to ${this.remotes}`);
+    const ismConfig = buildIsmConfigMap(owner, this.remotes, this.chains);
+    const ismContracts = await promiseObjAll(
+      objMap(ismConfig, async (chain, config) => {
+        return {
+          interchainSecurityModule: await ismFactory.deploy(chain, config),
```
Comment (jmrossy) [inline src/core/HyperlanePermissionlessDeployer.ts]: Optional: This structure allows for some parallelism but we may get better logging + error handling if we do them sequentially (e.g. if one chain fails, is it clear from logs which one?)

**Review by jmrossy — COMMENTED** [review]
Mostly looks good, just a few small optional things. I'd recommend we merge @tkporter's branch (#57) into here before merging. May as well update to the latest SDK.

**Review by jmrossy — APPROVED** [review]

**Pull request merged**